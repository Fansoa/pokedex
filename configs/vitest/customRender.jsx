import { render } from "@testing-library/react";
import IntegrationProvider from "@/configs/vitest/testUtils";

const renderWithIntegrationProvider = (children) =>
  render(<IntegrationProvider>{children}</IntegrationProvider>);

export default renderWithIntegrationProvider;
