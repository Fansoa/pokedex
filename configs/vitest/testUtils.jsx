import { ThemeProvider } from "@emotion/react";
import PropTypes from "prop-types";
import theme from "@/src/theme/theme";

const IntegrationProvider = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

IntegrationProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

export default IntegrationProvider;
