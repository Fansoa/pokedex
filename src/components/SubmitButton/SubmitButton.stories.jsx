import SubmitButton from "@/src/components/SubmitButton";
import { action } from "@storybook/addon-actions";

export default {
  title: "Components/SubmitButton",
  component: SubmitButton,
};

export const SubmitButtonStory = {
  name: "Default",
  args: {
    label: "Search !",
    onClick: action("Button is clicked"),
  },
};
