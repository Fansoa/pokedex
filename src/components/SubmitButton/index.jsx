import Button from "@/src/components/SubmitButton/styled";
import PropTypes from "prop-types";

const SubmitButton = ({ onClick, label }) => {
  return <Button onClick={onClick}>{label}</Button>;
};

SubmitButton.propTypes = {
  label: PropTypes.string,
  onClick: PropTypes.func,
};

SubmitButton.defaultProps = {
  label: null,
  onClick: null,
};

export default SubmitButton;
