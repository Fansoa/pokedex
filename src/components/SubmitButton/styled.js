import styled from "@emotion/styled";
import pxToRem from "@/src/utils/methods/pxToRem";

const Button = styled.button`
  ${({ theme }) => `
    background-color: ${theme.colors.sandyBrown};
    font-size: ${pxToRem(16)};
    font-weight: bold;
    width: ${pxToRem(156)};
    height: ${pxToRem(65)};
    border-radius: ${pxToRem(80)};
    color: ${theme.colors.white};
  `}
`;

export default Button;
