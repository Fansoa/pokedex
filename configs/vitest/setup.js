import { expect, afterEach } from "vitest";
import { cleanup } from "@testing-library/react";
import * as matchers from "@testing-library/jest-dom/matchers";
import { createSerializer } from "@emotion/jest";

expect.extend(matchers);
expect.addSnapshotSerializer(createSerializer());

afterEach(() => {
  cleanup();
});
