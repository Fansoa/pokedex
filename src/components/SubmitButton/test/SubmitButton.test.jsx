import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { expect, vi } from "vitest";
import renderWithIntegrationProvider from "@/configs/vitest/customRender";
import SubmitButton from "@/src/components/SubmitButton";
import { act } from "react-dom/test-utils";

describe("SubmitButton", () => {
  let container;

  it("onClick func is called when button is called", async () => {
    const onClickMock = vi.fn();
    container = renderWithIntegrationProvider(
      <SubmitButton onClick={onClickMock} label="label" />,
    ).container;
    const user = userEvent.setup();
    const button = screen.getByRole("button", {
      name: /label/i,
    });
    await act(async () => {
      await user.click(button);
    });
    expect(onClickMock).toHaveBeenCalledTimes(1);
    expect(container).toMatchSnapshot();
  });
});
