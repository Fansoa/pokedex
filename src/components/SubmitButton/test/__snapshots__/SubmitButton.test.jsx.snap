// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SubmitButton > onClick func is called when button is called 1`] = `
.emotion-0 {
  background-color: #F4A261;
  font-size: 1rem;
  font-weight: bold;
  width: 9.75rem;
  height: 4.0625rem;
  border-radius: 5rem;
  color: white;
}

<div>
  <button
    class="emotion-0"
  >
    label
  </button>
</div>
`;
